src/OrchardCore.Modules/OrchardCore.Contents/Liquid/ContentAnchorTag.cs

```diff
@@ -230,6 +214,13 @@ public async ValueTask<Completion> WriteToAsync(List<FilterArgument> argumentsLi
             tagBuilder.RenderEndTag().WriteTo(writer, (HtmlEncoder)encoder);
 
             return Completion.Normal;
+
+            async Task<ContentItemMetadata> PopulateAspectForContentItemMetadataAsync(ContentItem item)
+            {
+                contentItem = item;
+
+                return await contentManager.PopulateAspectAsync<ContentItemMetadata>(item);
```

Comment:
```suggestion
                return await contentManager.PopulateAspectAsync<ContentItemMetadata>(contentItem);
```

Reply:
Both are valid ;)

Reply:
But the idea is to create a new object so that the original one doesn't get morphed. Here, it is not consistent with what was done before if you use "item". It would be good to see if the original var is mutated too.

Reply:
If is that I will rid of the argument and assign it to `contentItem` directly, unless I did it for a reason

Reply:
It's for a reason

```
contentItem = createFor;
var metadata = await contentManager.PopulateAspectAsync<ContentItemMetadata>(createFor);
```

the `item` should has the `createFor` value

Reply:
Ok, I missed something here. You are defining an async task at the end of this method so it is not a different method. I understand now what you are saying.